### PR TITLE
test: Tests that validate transaction queue integrity for connection pool.

### PR DESCRIPTION
**Pull request by SandPod**
Tests that validate the issue described in: https://github.com/isoos/postgresql-dart/issues/297

Not meant to be merged but rather to help with describing and debugging the issue.

### CODE REVIEW

File: test/pool_test.dart [hunk 1]
```diff
@@ -61,6 +61,95 @@ void main() {
...
+      expect(firstResults, [
+        [1]
+      ]);
+      expect(secondResults, [
+        [1],
+        [2]
+      ]);
+    });
+
+    test(
```
Comment (SandPod) [inline test/pool_test.dart:106]: This test fails.

**Comment by isoos** [comment]
The test uses a setting where up to 8 concurrent connections are allowed:
```
        settings: PoolSettings(maxConnectionCount: 8),
```

When I add an extra id to the connections, I can see that the two `runTx` calls are given two different physical connections, and they are executed concurrently. If you change that setting to `maxConnectionCount: 1` you will get the expected desire.

I'm a bit unsure if we need to do anything here, as this is a kind of expected behavior to me.

**Comment by SandPod** [comment]
> I'm a bit unsure if we need to do anything here, as this is a kind of expected behavior to me.

I'm totally open to us misusing the framework, so please stay with me and clear out what I find strange :).

What confuses me is that the queue behaves differently depending on previous finalized requests.

In one of the connection pool test I don't perform any unawaited request before the transactions are initiated, and in that case the transactions are queued on the same connection with "my" wanted behavior.

While if we run at least 2 concurrent requests that are then awaited (so that all connections are still free if I understand it correctly), the transactions are queued on different connections.

What makes these scenarios behave differently, and is that what is expected?
Shouldn't the transactions always be queue either on the same or different connections?

**Comment by SandPod** [comment]
Not a valid test.

My question still stands around the queue scheduling.

**Comment by isoos** [comment]
> What makes these scenarios behave differently, and is that what is expected?

They are on different connections, and on the Dart side, there is no further assumption with them. As I've said before: there is no coordination of transaction on the Dart side, it is just a wrapper for opening and closing transactions using SQL commands. Everything is on the server how it receives and processes them.

**Pull request closed without merging**